### PR TITLE
A small set of edits that allow Heltec ESP32 target support

### PR DESCRIPTION
**Pull request by lokkju**
Primarily includes edits to make the RfEvent enum names unique (DIO0 is a widely used name) and a cmake toolchain file for compiling for Heltec ESP32 targets (using the Arduino board support package)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,10 +20,19 @@ Currently, only uplink is supported.
...
    ```
+   or to cross compile for Heltec ESP32
+   ```
+   PATH="$PATH:~/.arduino15/packages/Heltec-esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/" cmake -H. -Bbuild  -DCMAKE_TOOLCHAIN_FILE=toolchain-gcc-xtensa-esp32.cmake
+   ```
 1. Compile
    ```
    make -C build
    ```
+1. Copy built static libraries to Arduino LongFi
```
Comment (lthiery) [inline README.md]: This is a useful instruction, but I would rather include anything to do with "Arduino" into the longfi-arduino library. Including this here could make one think that the sole purpose of this library is to compile the static libraries for Arduino bindings.

File: README.md [hunk 2]
```diff
@@ -20,10 +20,19 @@ Currently, only uplink is supported.
    ```
    cmake -H. -Bbuild  -DCMAKE_TOOLCHAIN_FILE=../toolchain-gcc-arm-none-eabi.cmake -DCMAKE_FLAGS="-march=armv6s-m"
    ```
+   or to cross compile for Heltec ESP32
```
Comment (lthiery) [inline README.md]:
```suggestion
   or to cross compile for Heltec ESP32 (after [setting up the Heltec ESP32 toolchain](https://docs.espressif.com/projects/esp-idf/en/release-v3.0/get-started/linux-setup.html))
```

File: README.md [hunk 3]
```diff
@@ -20,10 +20,19 @@ Currently, only uplink is supported.
...
+   ```
+   PATH="$PATH:~/.arduino15/packages/Heltec-esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/" cmake -H. -Bbuild  -DCMAKE_TOOLCHAIN_FILE=toolchain-gcc-xtensa-esp32.cmake
```
Comment (lthiery) [inline README.md]:
```suggestion
 cmake -H. -Bbuild  -DCMAKE_TOOLCHAIN_FILE=toolchain-gcc-xtensa-esp32.cmake
```

File: longfi.c [hunk 4]
```diff
@@ -208,32 +208,32 @@ longfi_handle_event(LongFi_t * handle, RfEvent_t event)
...
-    case Timer1:
+    case RFE_Timer1:
         // TODO: needs to dispatch the callback stashed from TimerInit
         break;
-    case Timer2:
+    case RFE_Timer2:
         // TODO: needs to dispatch the callback stashed from TimerInit
         break;
-    case Timer3:
+    case RFE_Timer3:
```
Comment (lthiery) [inline longfi.c]: Good idea to rename

File: README.md (hunk 3 above)
Comment (lthiery) [reply README.md]: If this project is going to include specific build instructions for the ESP32, I'd rather it be Arduino agnostic, so I've made the two suggestions above. This has the benefit of not having implied instructions about installing the Arduino side of things, as was the current state.

**Comment by lokkju** [comment]
I think that should have resolved all of your comments; thanks!

**Pull request merged**